### PR TITLE
Updated constants from Thor into ACDC and updated the shooter subsystem

### PR DESCRIPTION
**Pull request by SamueljWilson**

### CODE REVIEW

**Review by ILIK3T0C0D3 — APPROVED** [review]
Looks great!

File: src/main/java/frc/robot/Constants.java [hunk 1]
```diff
@@ -95,42 +96,42 @@ public static class DriveConstants {
...
-      public static final int kTurningMotorCurrentLimit = 20;
-
-      public static final double kDriveMotorRampRate = 0.25;
-      public static final double kTurningMotorRampRate = 0.25;
-
-      public static final long kValueCacheTtlMicroseconds = 15;
+    public static final double kMaxSpeedMetersPerSecond = DriveConstants.kMaxSpeedMetersPerSecond * 2.0;
+
+    //TODO: Find actual wheel diameter
+    public static final double kWheelDiameterMeters = 0.09924;
```
Comment (jasone) [inline src/main/java/frc/robot/Constants.java:102]: Why did this number change if we still don't know the true value?

File: src/main/java/frc/robot/Constants.java [hunk 2]
```diff
@@ -209,14 +210,20 @@ public static class ShooterConstants {
...
-    public static final double kFlywheelEncoderCPR = 42.0;
-    public static final double kAbsoluteEncoderCPR = 4096.0;
-    //TODO: Find the flywheel diameter
-    public static final double kFlywheelDiameterMeters = 0.09525;
-    //TODO: Find gear ratio
-    public static final double kFlywheelGearRatio = 6.75;
-
-    public static final double kFlywheelEncoderDistancePerPulse =
-        (kFlywheelDiameterMeters * Math.PI) / (kFlywheelEncoderCPR * kFlywheelGearRatio);
+    public static final double kFlywheelDiameterMeters = Units.inchesToMeters(4.0) * Math.PI;
```
Comment (jasone) [inline src/main/java/frc/robot/Constants.java]: This is circumference, not diameter.

File: src/main/java/frc/robot/subsystems/Shooter.java [hunk 3]
```diff
@@ -56,39 +53,41 @@ public Shooter(int leftMotorID, int rightMotorID, boolean leftMotorReversed, boo
...
+    double leftMotorVelocity = m_leftEncoder.getVelocity();
     return (
-      m_rightMotor.get() <= m_currentRightSpeed + m_kShootingTolerance || 
-      m_rightMotor.get() >= m_currentRightSpeed - m_kShootingTolerance &&
-      m_leftMotor.get() <= m_currentLeftSpeed + m_kShootingTolerance || 
-      m_leftMotor.get() >= m_currentLeftSpeed - m_kShootingTolerance
+      rightMotorVelocity <= m_idealRightSpeed + m_kShootingTolerance || 
+      rightMotorVelocity >= m_idealRightSpeed - m_kShootingTolerance &&
+      leftMotorVelocity <= m_idealLeftSpeed + m_kShootingTolerance || 
+      leftMotorVelocity >= m_idealLeftSpeed - m_kShootingTolerance
```
Comment (jasone) [inline src/main/java/frc/robot/subsystems/Shooter.java]: Addition doesn't work here. You need to be multiplying/dividing.

**Review by jasone — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/Constants.java (hunk 1 above)
Comment (SamueljWilson) [reply src/main/java/frc/robot/Constants.java:102]: It is the measured ideal diameter, during our conversation you suggested a better way to find diameter, so this is just an ideal diameter for right now.

File: src/main/java/frc/robot/subsystems/Shooter.java [hunk 4]
```diff
@@ -77,11 +77,14 @@ public void stopShooter() {
...
+    double leftTolerance = m_idealRightSpeed * m_kShootingTolerance;
     return (
-      rightMotorVelocity <= m_idealRightSpeed + m_kShootingTolerance || 
-      rightMotorVelocity >= m_idealRightSpeed - m_kShootingTolerance &&
-      leftMotorVelocity <= m_idealLeftSpeed + m_kShootingTolerance || 
-      leftMotorVelocity >= m_idealLeftSpeed - m_kShootingTolerance
+      rightMotorVelocity * (1.0 - rightTolerance) <= m_idealRightSpeed || 
+      rightMotorVelocity * (1.0 + rightTolerance) >= m_idealRightSpeed &&
+      leftMotorVelocity  * (1.0 -leftTolerance) <= m_idealLeftSpeed || 
+      leftMotorVelocity  * (1.0 + leftTolerance) >= m_idealLeftSpeed
```
Comment (jasone) [inline src/main/java/frc/robot/subsystems/Shooter.java]: This won't return true unless both motor velocities are outside the desired range.

**Review by jasone — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/subsystems/Shooter.java [hunk 5]
```diff
@@ -28,7 +28,7 @@ public class Shooter extends SubsystemBase {
   PID m_leftMotorPID = ShooterConstants.kLeftPID;
   PID m_rightMotorPID = ShooterConstants.kLeftPID;
 
-  double m_kShootingTolerance = 0.127;
+  double m_kShootingTolerance = 0.05;
```
Comment (jasone) [inline src/main/java/frc/robot/subsystems/Shooter.java]: This should be in `Constants.java`.

**Review by jasone — CHANGES_REQUESTED** [review]

**Review by jasone — APPROVED** [review]

**Pull request merged**